extensions/transport/portability-transport-jettyrest/src/main/java/org/datatransferproject/transport/jettyrest/http/JettyTransport.java

```diff
@@ -116,6 +120,58 @@ public void registerServlet(String path, Servlet servletContainer) {
 
   private class JettyErrorHandler extends ErrorHandler {
 
+    /**
+     * Hack to intercept any non-"api/*" requests that make it to the API server. This
+     * includes:
+     * - "/" -> index.html
+     * - "/callback?token=foo" -> index.html?token=foo
+     *
+     * <p>This is NOT used in the demo-server since nginx handles these requests with
+     * the Angular client.
+     *
+     * <p>This IS used by the demo-google-deployment, as the Google Cloud Load Balancer
+     * currently used in that deployment can't do URL remaps - so we solve that by setting
+     * our default mapping to the API backends and redirecting those requests here.
+     * Otherwise "/" will map to our static bucket and display a directory listing rather
+     * than index.html.
+     *
+     * TODO(rtannenbaum): Move this somewhere better than the error handler
+     */
+    @Override
+    public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response)
+            throws IOException
+    {
+      logger.warn("handling non-'api/*' request: " + request.getServletPath()
+              + ". Do NOT do this in production environments!");
+      if (request.getServletPath().equals("/")) {
+        String redirect = request.getServerName() + "/index.html";
+        redirect = (request.isSecure() || !request.getServerName().equals("localhost") ? "https://" : "http://")
```

Comment:
Can use use the useHttps variable to drive this rather than having different logic that determines if the api is running with https?

Reply:
Outdated, we don't need this hack in the API any more with the new "website" static bucket.